src/components/Rating/Rating.tsx

```diff
@@ -0,0 +1,330 @@
+import React, { Component } from 'react'
+import CSS from "csstype";
+
+export interface propTypes {
+  className?: string;
+  edit?: boolean;
+  half?: boolean;
+  value: number;
+  count: number;
+  char?: string;
+  size: number;
+  color1?: string;
+  color2?: string;
+  onChange: any;
+  labels: string[];
+  labelIndex: number;
+}
+
+interface stateTypes {
+  uniqueness?: string;
+  value: number;
+  stars: {active: boolean}[];
+  halfStar: {
+    at?: number;
+    hidden?: boolean;
+  };
+  config: {
+    count: number;
+    size: number;
+    char?: string;
+    // default color of inactive star
+    color1?: string;
+    // color of an active star
+    color2?: string;
+    half?: boolean;
+    edit?: boolean;
+  };
+  labelIndex: number;
+}
+
+const parentStyles: CSS.Properties = {
+  overflow: 'hidden',
+  // position: 'relative'
+}
+
+const defaultStyles: CSS.Properties = {
+  position: 'relative',
+  // overflow: 'hidden',
+  cursor: 'pointer',
+  display: 'block',
+  float: 'left'
+}
```

Comment:
I think it may be better to add transition to make hovering animation effect smooth. Also by adding `align-items:center` can make the stars in straight line when hovering.

Reply:
done